src/DynamoUtilities/PIIDetector.cs

```diff
@@ -0,0 +1,125 @@
+                    var annotations = viewProperties.FirstOrDefault(x => x.Name == "Annotations");
+
+                    if (annotations != null)
+                    {
+                        if (annotations.Value.Type == JTokenType.Array)
+                        {
+                            var annotationsList = (JArray)annotations.Value;
+
+                            foreach (JObject annotation in annotationsList)
+                            {
+                                var annotationProperties = annotation.Children<JProperty>();
+                                JProperty title = annotationProperties.FirstOrDefault(x => x.Name == "Title");
+                                if (title != null) { title.Value = RemovePIIData((string)title.Value); }
+                            }
+                        }
+                    }
+                }
+            }
+            return jObjectResult;
+        }
+
+        static string emailPattern = @"\w+([-+.]\w+)*@\w+([-.]\w+)*\.\w+([-.]\w+)*";
+        static string websitePattern = @"(?:http(s)?:\/\/)?[\w.-]+(?:\.[\w\.-]+)+[\w\-\._~:/?#[\]@!\$&'\(\)\*\+,;=.]+";
+        static string directoryPattern = @"(^([a-z]|[A-Z]):(?=\\(?![\0-\37<>:""/\\|?*])|\/(?![\0-\37<>:""/\\|?*])|$)|^\\(?=[\\\/][^\0-\37<>:""/\\|?*]+)|^(?=(\\|\/)$)|^\.(?=(\\|\/)$)|^\.\.(?=(\\|\/)$)|^(?=(\\|\/)[^\0-\37<>:""/\\|?*]+)|^\.(?=(\\|\/)[^\0-\37<>:""/\\|?*]+)|^\.\.(?=(\\|\/)[^\0-\37<>:""/\\|?*]+))((\\|\/)[^\0-\37<>:""/\\|?*]+|(\\|\/)$)*()";
+        static string creditCardPattern = @"(\d{4}[-, ]\d{4})";
+        static string ssnPattern = @"\d{3}[- ]\d{2}[- ]\d{4}";
+        static string ipPattern = @"((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)";
+        static string datePattern = @"\d{1,2}[/-]\d{1,2}[/-]\d{2,4}";
+
+        public static JToken GetNodeById(JObject jsonWorkspace,string nodeId)
+        {
+            return jsonWorkspace["Nodes"].Where(t => t.Value<string>("Id") == nodeId).Select(t => t).FirstOrDefault();
+        }
+
+        public static JToken GetNodeValue(JObject jsonWorkspace, string nodeId,string propertyName)
+        {
+            var node = jsonWorkspace["Nodes"].Where(t => t.Value<string>("Id") == nodeId).Select(t => t).FirstOrDefault();
+            var property = node.Children<JProperty>().FirstOrDefault(x => x.Name == propertyName);
+            return property.Value;
+        }
+
+        public static JToken GetNoteValue(JObject jsonWorkspace, string nodeId)
+        {
+            var x = jsonWorkspace["View"]["Annotations"];
+            var note = jsonWorkspace["View"]["Annotations"].Where(t => t.Value<string>("Id") == nodeId).Select(t => t).FirstOrDefault();
+            var property = note.Children<JProperty>().FirstOrDefault(x => x.Name == "Title");
+            return property.Value;
+        }
+
+        public static bool ContainsEmail(string value) { return new Regex(emailPattern).Match(value).Success; }
```

Comment:
be conservative as possible with access modifiers, none of this should be something we are providing as APIs - right?

Reply:
Done.